### DISCUSSION TITLE
VS Code (code-server) on Google Colab / Kaggle Notebooks / Anywhere

### DISCUSSION THREAD

**Discussion by abhishekkrthakur** [coder/code-server#2084] [Show and tell]
I made a small library/tool that one can use to have code-server on any machine which is connected to the internet (inspired by a recent article). I call it ColabCode. This is a small video I made demonstrating that: https://www.youtube.com/watch?v=7kTbM3D02jU

The repository is available here: https://github.com/abhishekkrthakur/colabcode

Quickstart:
- install colabcode: `pip install colabcode`
- import colabcode: `from colabcode import ColabCode`
- run: `ColabCode(port=10000, password="abhishek")`

Please let me know how you like it :)

**Comment by bpmct** [comment]
This is so cool! Definitely one of the quickest ways to start a code-server environment with https! I love how you preinstalled some extensions too.
 
![Notebooks GIF](https://user-images.githubusercontent.com/22407953/92925610-bdd52680-f408-11ea-81a8-3818e7b4f1ba.gif)

**Reply by abhay-ranawat** [reply]
Can we use [https://tunnelto.dev/](tunnelto) insted on ngork

**Reply by bpmct** [reply]
@abhay-ranawat Yeah, it seems totally possible. You would just have to modify it in https://github.com/abhishekkrthakur/colabcode/blob/master/colabcode/code.py and install tunnel, and run the tunnel command as a subprocess instead of utilizing the ngork library.
